### ISSUE TITLE
MapKit Globe view camera resets when mov

### ISSUE THREAD

**Issue by subtleGradient** [react-native-maps/react-native-maps#4880]
### Summary

When viewing a map in globe view, occasionally the camera/region get reset back to 0.

Here I'm trying to drag the north pole into view, but when I get too far north it resets back out
![Simulator Screen Recording - 13 Mini - 2023-11-06 at 15 17 35](https://github.com/react-native-maps/react-native-maps/assets/4117/dd88b37a-7919-47c6-bae7-20e15c60c23f)

looks like it's related to this code:
https://github.com/react-native-maps/react-native-maps/blob/4514bd13a34e20f7e82e7de0b44224b6160195d5/ios/AirMaps/AIRMapManager.m#L1167-L1174

### Reproducible sample code

```TSX
import { Platform } from "react-native";
import MapView from "react-native-maps";

export function DevMap_camResetBugRepro() {
  const mapType =
    Platform.OS === "ios"
      ? ("hybridFlyover" as any) // NOTE: casting to any because this is not in the type definition, but it works to set the map type to a 3D globe
      : "standard";

  return (
    <MapView
      style={{ flex: 1 }}
      mapType={mapType}
      initialRegion={{
        latitude: 14,
        longitude: 30,
        latitudeDelta: 156,
        longitudeDelta: 173,
      }}
    />
  );
}
```

### Steps to reproduce

Drag the north pole into view

### Expected result

see the north pole

### Actual result

region/camera resets back out to latitude 0

### React Native Maps Version

1.7.1

### What platforms are you seeing the problem on?

iOS (Apple Maps)

### React Native Version

0.72.3

### What version of Expo are you using?

SDK 48

### Device(s)

iPhone 13 mini

**Comment by subtleGradient** [comment]
also related: https://github.com/react-native-maps/react-native-maps/blob/4514bd13a34e20f7e82e7de0b44224b6160195d5/ios/AirMaps/AIRMapManager.m#L1090-L1093

**Comment by subtleGradient** [comment]
looks like this is the code that triggers the zoom out:
https://github.com/react-native-maps/react-native-maps/blob/4514bd13a34e20f7e82e7de0b44224b6160195d5/ios/AirMaps/AIRMapManager.m#L951-L956

**Comment by subtleGradient** [comment]
Workaround:
```ts
```

**Comment by jan-kozinski** [comment]
Hi, that's because the calculation of zoom level doesn't take into account the map's rotation. I'll try to find a better algorithm

**Comment by salah-ghanim** [comment]
@jan-kozinski yes this seems like a clear bug ... specifically apple maps code was introduced before 3d maps were available.

there are two ways to could go about
1. move away from this algorithem to apple's introduced method setCameraZoomRange 
https://developer.apple.com/documentation/mapkit/mkmapview/3114302-setcamerazoomrange?language=objc
supported since iOS 13, but requires rethinking the logic since it's in meters 

2. simply disable max/min zoom logic for globe view until we implement 1 correctly.

**Comment by jan-kozinski** [comment]
The problem lays a bit deeper than just the map rotation. The hybrid flyover map is a globe without mercator projection. The cuurent zoom counting algo is designed for flat maps projected with mercator. I can't find any formula for globe zoom or even scale. 

Thanks for advice @salah-ghanim , I think we can use camera zoom range approach for hybidFlyover and satteliteFlyover maps, with min camera zoom range in meters being 2^minZoomLevel, max being being 2^maxZoomLevel and leave the original algorithm for other map types. I'll put up a PR

**Comment by jan-kozinski** [comment]
Hi @subtleGradient I made a fix so you might want to see if that addresses your issue in PR #4892

**Comment by ahinkle** [comment]
For what it's worth, it also happens on the South Pole. Bummer
